task/control-plane-agent/src/mgs_sidecar.rs

```diff
@@ -166,19 +238,236 @@ impl MgsHandler {
 
         Ok(state)
     }
+
+    /// Unlocks the tech port if the challenge and response are compatible
+    fn unlock(
+        &mut self,
+        vid: u16,
+        challenge: UnlockChallenge,
+        response: UnlockResponse,
+        time_sec: u32,
+    ) -> Result<(), GwMonorailError> {
+        ringbuf_entry!(Trace::UnlockRequested(vid));
+
+        if time_sec > MAX_UNLOCK_TIME_SECS {
```

Comment:
We may want the signed hash to also cover the unlock duration.
e.g. when more sophisticated signing policies are implemented then a customer may be fine with unlock for 10 minutes but not MAX_UNLOCK_TIME_SECS.